qstash/workflow/migration.mdx

```diff
@@ -0,0 +1,89 @@
+---
+title: "Migration to New SDK"
+---
+
+On October 18, 2024 we released a new sdk `@upstash/workflow` for Upstash Workflow. Previously,
+`@upstash/qstash` was used. `@upstash/qstash` will continue being developed alongside QStash,
+but future Upstash Workflow development will take place in `@upstash/workflow`. Workflow
+imports will be removed from `@upstash/qstash` in a future release.
+
+If you had started using Upstash Workflow with `@upstash/qstash`, you will
+need to migrate to `@upstash/workflow`. We have made some backwards incompatible changes but we
+want to make the transition as smooth as possible.
+
+In this guide, we will explain the changes you may need to make for migration.
+
+### Install `@upstash/workflow`
+
+First, we will need to install the new package with:
+
+```
+npm install @upstash/workflow
+```
+
+If you were using `@upstash/qstash` only for workflow, you can uninstall it from your project.
+
+### Serve methods
+
+Firstly, you will need to change the imports from `@upstash/qstash` to @upstash/workflow:
+
+```ts
+// old
+import { serve } from "@upstash/qstash/nextjs"
+
+// new 
+import { serve } from "@upstash/workflow/nextjs"
+```
+
+We have updated what our `serve` methods return. We made this change to make it
+easier to extend the API in the future.
+
+For instance, Next.js method changed like this:
+
+```javascript
+// old
+export const POST = serve(...);
+
+// new
+export const { POST } = serve(...);
+```
+
+We kept the `serve` method of `Hono` the same. The rest are updated in a similar way.
+See [the quickstarts](/qstash/workflow/quickstarts/platforms) for the new way `serve`
+should be used.
+
+Additionally, `@upstash/workflow/nuxt` import is removed. You should use `@upstash/workflow/h3`
```

Comment:
Just to keep myself updated, why did we do this?

Reply:
When I first added nuxt handler, i added it under upstash/qstash/nuxt
[But nuxt uses h3 under the hood](https://nuxt.com/docs/guide/concepts/server-engine), so the handler wasn’t specific to nuxt
So we moved the nuxt handler to upstash/qstash/h3 but kept upstash/qstash/nuxt for backward compatibility
Now I removed it

Reply:
Maybe you can add a small note below explaining this.